Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py

```diff
@@ -0,0 +1,370 @@
+
+            # Supply the collection to TAXIICollection
+            tc_source = TAXIICollectionSource(collection_data)
+
+            filter_objs = {
+                "Technique": {"name": "attack-pattern", "filter": Filter("type", "=", "attack-pattern")},
+                "Mitigation": {"name": "course-of-action", "filter": Filter("type", "=", "course-of-action")},
+                "Group": {"name": "intrusion-set", "filter": Filter("type", "=", "intrusion-set")},
+                "Malware": {"name": "malware", "filter": Filter("type", "=", "malware")},
+                "Tool": {"name": "tool", "filter": Filter("type", "=", "tool")},
+                "relationships": {"name": "relationships", "filter": Filter("type", "=", "relationship")},
+            }
+
+            for concept in filter_objs:
+                if 0 < limit <= counter:
+                    break
+
+                input_filter = filter_objs[concept]['filter']
+                try:
+                    mitre_data = tc_source.query(input_filter)
+                except Exception:
+                    continue
+
+                for mitre_item in mitre_data:
+                    if 0 < limit <= counter:
+                        break
+
+                    mitre_item_json = json.loads(str(mitre_item))
+                    if mitre_item_json.get('id') not in mitre_id_list:
+                        value = mitre_item_json.get('name')
+                        indicator_type = MITRE_TYPE_TO_DEMISTO_TYPE.get(mitre_item_json.get('type'))  # type: ignore
+
+                        if indicator_type == 'Relationship' and create_relationships:
+                            if mitre_item_json.get('relationship_type') == 'revoked-by':
+                                continue
+                            relation_obj = create_relationship(mitre_item_json, id_to_name)
+                            relationships_list.append(relation_obj.to_indicator()) if relation_obj else None
+
+                        else:
+                            if is_indicator_deprecated_or_revoked(mitre_item_json):
+                                continue
+                            id_to_name[mitre_item_json.get('id')] = value
+                            indicator_score = INDICATOR_TYPE_TO_SCORE.get(indicator_type)  # type: ignore
+                            indicator_obj = {
+                                "value": value,
+                                "score": indicator_score,
+                                "type": indicator_type,
+                                "rawJSON": mitre_item_json,
+                                "fields": map_fields_by_type(indicator_type, mitre_item_json)  # type: ignore
+                            }
+
+                            indicator_obj['fields']['tags'].extend(self.tags)
+
+                            if self.tlp_color:
+                                indicator_obj['fields']['trafficlightprotocol'] = self.tlp_color
```

Comment:
this can be a separate function of indicator creation.

Reply:
done